src/runtime_src/core/pcie/emulation/common_em/config.cxx

```diff
@@ -980,4 +981,61 @@ namespace xclemulation{
     }
   }
 
+  int getIPName2Index(const char *name, xclBin* header)
+  {
+    std::string errmsg;
+    const uint64_t bad_addr = 0xffffffffffffffff;
+    char* ipLayoutbuf = nullptr;
+    size_t bufsize = 0;
+
+    if (!header) {
+      errmsg = "ERROR: getIPName2Index - Invalid xclbin content";
+      std::cout << errmsg << std::endl;
+      return -EINVAL;
+    }
+
+    char *bitstreambin = reinterpret_cast<char*> (header);
+    auto top = reinterpret_cast<const axlf*>(header);
+    if (auto sec = xclbin::get_axlf_section(top, IP_LAYOUT)) {
+      bufsize = sec->m_sectionSize;
+      ipLayoutbuf = new char[bufsize];
```

Comment:
Where is `ipLayoutbuf `deleted?  You should use `std::vector<char>` for the buffer to automatically manage lifetime.

But why make a copy of  the IP_LAYOUT section in the first place, it's just an offset into the xclbin buffer.

Reply:
@stsoe, Removed code for IP_LAYOUT copy to ipLayoutbuf